opal/corelib/array.rb

```diff
@@ -63,7 +76,15 @@ def initialize(size = nil, obj = nil, &block)
   end
 
   def self.new(*args, &block)
-    [].initialize(*args, &block)
+    obj = allocate
+    breaker = obj.initialize(*args, &block)
+    %x{
+      if (obj.$$has_breaker !== undefined) {
```

Comment:
@iliabylich LOL, that was [my question there as well](https://github.com/opal/opal/pull/1246/files#r47440335). If I understand correctly, you are checking the return value of `initialize`, and since it normally should be the object itself, you can *assume* that if it's *anything else*, then it *must* be, can *only ever be*, `$breaker.$v`. Right? That's basically what I'm doing with the return value of `String#upto` called from `Range#each`, but I have no idea if that's "the right way". Actually, there may not be a right way, given that the semantics of `break` appear to be [broken](https://github.com/opal/opal/issues/1247) at this time. ¯\_(ツ)_/¯